### PR TITLE
RTL fixes

### PR DESCRIPTION
**Pull request by yosilevy**
Kudos on this great card. Started using it today. Like much of HA it lacks RTL support which I have been slowly fixing.

Changes include:
- Using ha-slider which I created quite a few HA versions ago (in the main repo) to fix slider pin problems.
- Settings directions where needed to keep controls functioning (otherwise slider works the wrong way and pin text is reversed).
- Style updates so components looks right.

Tested in both LTR and RTL to make sure nothing broken.

Before fix (see highlights for problem):
image

After fix:
Capture

### CODE REVIEW

**Comment by kalkih** [comment]
Oh, I actually wasn't aware that the whole UI is flipped when using an RTL locale.

Not sure if I wanna switch from paper-slider to ha-slider as it will break for people running older versions of HA where ha-slider isn't available.

The control strip (volume slider & media buttons) looks weird in your RTL screenshots, it should probably be left aligned.

**Comment by yosilevy** [comment]
Fair enough. Reverted back to paper-slider + added patch to the style (they have a known bug and this is how we fixed it in HA main repo).
Went through all styles and added [rtl] styles to match.

Looks great now.

image

**Comment by kalkih** [comment]
Awesome, that looks much better indeed!

File: src/style.js [hunk 1]
```diff
@@ -465,7 +490,7 @@ const style = css`
   .speaker-select > paper-checkbox > span {
     font-weight: 600;
   }
-  paper-slider {
```
Comment (kalkih) [inline src/style.js]: Need to revert all these selectors back to paper-slider

File: src/main.js [hunk 2]
```diff
@@ -173,6 +175,7 @@ class MiniMediaPlayer extends LitElement {
         artwork=${config.artwork} ?has-artwork=${artwork} state=${this.player.state}
         ?flow=${config.flow} ?collapse=${config.collapse}
         content=${this.player.content}
+        ?rtl=${this._computeRTL(this.hass)}
```
Comment (kalkih) [inline src/main.js]: What do you think about making this check once inside `set hass()` instead and save the result as a boolean in a member variable (this.rtl) instead? Then we could resure that variable throughout the rendering logic. Like below.

https://github.com/kalkih/mini-media-player/blob/bc848c6e59cc100e8f8b661b036ae552df148b8a/src/main.js#L57-L61

File: src/main.js [hunk 3]
```diff
@@ -488,6 +503,7 @@ class MiniMediaPlayer extends LitElement {
           @change=${e => this.player.setVolume(e)}
           @click=${e => e.stopPropagation()}
           min='0' max=${this.config.max_volume} value=${this.player.vol * 100}
+          dir=${this._computeRTLDirection(this.hass)}
```
Comment (kalkih) [inline src/main.js]: If we take the member variable approach, we could do 
```js
?rtl=${this.rtl}
```
here. Would have to change the slider patch slightly for the selector to work.

File: src/main.js [hunk 4]
```diff
@@ -148,6 +148,8 @@ class MiniMediaPlayer extends LitElement {
     ro.observe(this.shadowRoot.querySelector('.player'));
     setTimeout(() => this.initial = false, 250);
     this.edit = this.config.speaker_group.expanded || false;
+
+    this.patchSliderForRTL();
```
Comment (kalkih) [inline src/main.js]: We only really need to patch this if rtl is `true`
```js
if (this.rtl) this.patchSliderForRTL();
```

File: src/style.js (hunk 1 above)
Comment (yosilevy) [reply src/style.js]: Please look again. All references have already been reverted. I think you looked at the wrong file.

Comment (kalkih) [reply src/style.js]: The HTML yes, but not the CSS selectors.

Comment (yosilevy) [reply src/style.js]: Where do you see in the CSS ha-paper-slider? They are removed.

Comment (kalkih) [reply src/style.js]: ha-paper-slider? You changed some CSS selectors to ha-slider, you have to revert those back to paper-slider, I'll make comments on them for you.

File: src/style.js [hunk 5]
```diff
@@ -410,7 +431,11 @@ const style = css`
     width: auto;
     max-width: 100%;
   }
-  .control-row--top paper-slider {
+  ha-card[rtl] .control-row--top {
+    margin-right: auto;
+    margin-left: 0;
+  }
+  .control-row--top ha-slider {
```
Comment (kalkih) [inline src/style.js]: Should be paper-slider

File: src/style.js [hunk 6]
```diff
@@ -465,7 +490,7 @@ const style = css`
   .speaker-select > paper-checkbox > span {
     font-weight: 600;
   }
-  paper-slider {
+  ha-slider {
```
Comment (kalkih) [inline src/style.js]: Should be paper-slider

File: src/style.js [hunk 7]
```diff
@@ -570,26 +595,38 @@ const style = css`
...
     margin-left: auto;
   }
+  ha-card[flow][rtl] .power-row {
+    margin-right: auto;
+  }
   ha-card[flow] .entity__info {
     display: none;
   }
-  ha-card[flow] paper-slider,
+  ha-card[flow] ha-slider,
```
Comment (kalkih) [inline src/style.js]: Should be paper-slider

File: src/style.js (hunk 1 above)
Comment (yosilevy) [reply src/style.js]: Yes. Sorry. Got confused with something else I was doing in HA main repo. Sorry.

**Comment by yosilevy** [comment]
All fixed as you suggested. Works fine.

File: src/style.js (hunk 1 above)
Comment (kalkih) [reply src/style.js]: No worries, are there both a ha-paper-slider and a ha-slider in core?

**Comment by yosilevy** [comment]
Was almost working fine. It wouldn't patch the slider if it was rendered after the card was already rendered. Added it to update method + marked as patched or not using classes.

File: src/style.js (hunk 1 above)
Comment (yosilevy) [reply src/style.js]: Yes unfortunately

**Comment by yosilevy** [comment]
Anything else pending?

File: src/main.js [hunk 8]
```diff
@@ -244,6 +255,18 @@ class MiniMediaPlayer extends LitElement {
...
+  _computeRTL(hass) {
+    const lang = hass.language || "en";
+    if (hass.translationMetadata.translations[lang]) {
+      return hass.translationMetadata.translations[lang].isRTL || false;
+    }
+    return false;
+  }
+
+  _computeRTLDirection(hass) {
+    return this._computeRTL(hass) ? "rtl" : "ltr";
```
Comment (kalkih) [inline src/main.js]: No need to compute it again, we got it in `this.rtl`

**Comment by kalkih** [comment]
My last comment + fixing the linter errors, rest looks good!

**Comment by yosilevy** [comment]
Do you know why code formatting doesn't work in this project? Does it work for you? That's why I didn't notice those issues.

**Comment by kalkih** [comment]
Not sure, should work fine as you can see in the Travis build. Run `npm run lint` and see what error you get.

**Comment by yosilevy** [comment]
Please let me know once you merge and release a version. Some RTL consumers are waiting for it... :-)

**Comment by kalkih** [comment]
Merged, will be included in the next release.

Great job, thanks for the PR! 🎉

**Pull request merged**